nfsv4/nfs4.h

```diff
@@ -1,15 +1,38 @@
-#include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <arpa/inet.h>
 #include <buffer.h>
+#include <vector.h>
 
-typedef struct server *server;
-typedef struct rpc *rpc;
+#define true (1)
+#define false (0)
+typedef int boolean;
 
-rpc allocate_rpc(server s); 
-u64 file_size(server s, char *pathname);
-void nfs4_connect(server);
-server create_server(char *hostname);
-void rpc_send(rpc);
-void readfile(server s, char *pathname, void *dest, u64 offset, u32 length);
+typedef struct status *status;
+typedef struct file *file;
+typedef struct client *client;
+
+status create_client(char *hostname, client *dest);
+
+status file_open_read(client c, vector path, file *x);
+status file_open_write(client c, vector path, file *x);
+status file_create(client c, vector path, file *x);
+void file_close(file f);
+status file_size(file f, u64 *s); // should be path?
```

Comment:
Maybe we could have both eventually, as in `stat` and `fstat`.

Reply:
yeah,certainly. this is another potential use of dictionaries. in general need to write some general routines for generating and reading fattr. there is also a general name/value facility on the server which might* be useful for inter-instance coordination if we can use the values directly in compound rocs